dips/dip-158.md

```diff
@@ -0,0 +1,794 @@
+---
+dip: 158
+title: P2M (eCommerce) Payments Off-chain API
+authors: Daniel Prinz, Yinon Zohar, Kevin Hurley (@kphfb)
+status: Draft
+type: Standard
+created: 03/15/2021
+---
+
+# Summary
+
+This DIP is an extension of the Off-Chain protocol for enabling eCommerce P2M payments (Peer To Merchant) of the following types:
+* Charge
+* Auth / Capture
+ 
+This DIP extends [DIP-1](https://dip.diem.com/dip-1/) which specifies P2P payment flows (Peer To Peer).
+
+This DIP replaces the Auth/Capture section in [DIP-8](https://dip.diem.com/dip-8/#authcapture).
+
+This DIP does NOT include payment flows which require funds pull pre-approval such as subscription and recurring payments. 
+
+This DIP makes use of the `result` field added to the `CommandResponseObject` as described in DIP-10
```

Comment:
Looks like we will have a DIP dedicated to the result field extension, DIP-161 (https://github.com/diem/dip/pull/165)

Reply:
Will add. Thanks for letting me know.